### ISSUE TITLE
Fails to build with GCC 11: error: invalid use of incomplete type 'class File' 

### ISSUE THREAD

**Issue by edo0** [jwilk/pdf2djvu#146]
Hello there!
I am maintaining the pdf2djvu package for Arch's AUR.
It's been [reported](https://aur.archlinux.org/packages/pdf2djvu/) to me that compilation fails using GCC11 on your latest official release from 2020 (pdf2djvu-0.9.17.1)
I was wondering if you could make a patched re-release, unless a new version altogether was on its way.
Thank you in advance

**Comment by jwilk** [comment]
Thanks for the bug report.

Indeed, building with GCC 11 fails:
```
In file included from debug.hh:23,
                 from config.cc:27:
system.hh:144:9: error: invalid use of incomplete type 'class File'
  144 |   File::streamoff size();
      |         ^~~~~~~~~
system.hh:127:7: note: definition of 'class File' is not complete until the closing brace
  127 | class File : public std::fstream
      |       ^~~~
```

This is fixed by d88c2307752269525b326601d1da177fcf796784.

I'll make a new release soon; most likely tomorrow.

**Comment by edo0** [comment]
Thank you very much, that would be great

**Comment by jwilk** [comment]
Fixed in [0.9.18](https://github.com/jwilk/pdf2djvu/releases/tag/0.9.18).
